src/common/viz/FigureExtractor.js

```diff
@@ -50,11 +58,15 @@ define(['./Utils'], function (Utils) {
             title: node.getAttribute('title'),
         };
 
-        let childrenIds = node.getChildrenIds();
+        let childrenIds = this.getMetadataChildrenIds(node);
+
         let childNode, childNodeFn;
         desc.subGraphs = childrenIds.map((childId) => {
             childNode = this._client.getNode(childId);
             childNodeFn = this.getMetaType(childNode);
+            if(!canExtractMetadata(childNodeFn)){
+                throw new Error(`Node of type ${childNodeFn} is not supported yet.`);
```

Comment:
It might be nice to change `canExtractMetadata` to something like `ensureCanExtract` where it throws the unsupported error. This would prevent this small amount of code duplication but more importantly would ensure that the error message is consistent. Otherwise, it is easy to update one and miss the other, etc.

Reply:
This has been addressed.